vslib/saivs.h

```diff
@@ -56,6 +56,16 @@ extern "C" {
  */
 #define SAI_KEY_VS_HOSTIF_USE_TAP_DEVICE      "SAI_VS_HOSTIF_USE_TAP_DEVICE"
 
+/**
+ * @def SAI_KEY_VS_OPER_SPEED_IS_CONFIGURED_SPEED
+ *
+ * Bool flag, (true/false). If set to true, SAI_PORT_ATTR_OPER_SPEED returns
+ * the SAI_PORT_ATTR_SPEED instead of querying the speed of veth
+ *
+ * By default this flag is set to false.
+ */
+#define SAI_KEY_VS_OPER_SPEED_IS_CONFIGURED_SPEED "SAI_VS_OPER_SPEED_IS_CONFIGURED_SPEED"
```

Comment:
this name is a little bit confusing, it suggest that this is a question or value to read and not an config value

why we need this actually ?

also does this only applies when tap device is in use?

Reply:
> this name is a little bit confusing,

How about `SAI_KEY_VS_USE_CONFIGURED_SPEED_AS_OPER_SPEED`?

> why we need this actually ?

I'm testing DPB on the VS environment.
To check that DPB is working as expected, it'd be nice to see the `show interface status` command shows the expected interface speed.

```
admin@sonic:~$ show interface status
  Interface    Lanes    Speed    MTU    FEC    Alias    Vlan    Oper    Admin                                             Type    Asym PFC
-----------  -------  -------  -----  -----  -------  ------  ------  -------  -----------------------------------------------  ----------
  Ethernet0  1,2,3,4     400G   9100    N/A   Eth0/0  routed    down     down  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
admin@sonic:~$ sudo config interface breakout Ethernet0 "2x200G"
Do you want to Breakout the port, continue? [y/N]: y

Running Breakout Mode : 1x400G 
Target Breakout Mode : 2x200G

Ports to be deleted : 
 {
    "Ethernet0": "400000"
}
Ports to be added : 
 {
    "Ethernet0": "200000",
    "Ethernet2": "200000"
}
Breakout process got successfully completed.
Please note loaded setting will be lost after system reboot. To preserve setting, run `config save`.
admin@sonic:~$ show interface status
  Interface    Lanes    Speed    MTU    FEC    Alias    Vlan    Oper    Admin                                             Type    Asym PFC
-----------  -------  -------  -----  -----  -------  ------  ------  -------  -----------------------------------------------  ----------
  Ethernet0      1,2     200G   9100    N/A   Eth0/0  routed    down     down  QSFP-DD Double Density 8X Pluggable Transceiver         N/A
  Ethernet2      3,4     200G   9100    N/A   Eth0/1  routed    down     down                                              N/A         N/A
```

> also does this only applies when tap device is in use?

I don't think so.  [`SwitchStateBase::refresh_port_oper_speed`](https://github.com/sonic-net/sonic-sairedis/blob/master/vslib/SwitchStateBase.cpp#L2246) doesn't care `SAI_VS_HOSTIF_USE_TAP_DEVICE` configuration.

Reply:
Changed name looks good